[Feature] Add the ability to disable teleport safety.

kashike opened the pull request:
Allow teleport safety to be configurable.
Requested by a few people in #essentials due to problems with other plugins (Factions for example) where players can get past walls, etc thanks to teleport safety.
Default configuration enables teleport safety, and there is a "It's recommended to leave this enabled." notice with the configuration option.

Carlgo11 commented:
Great work! We really needed such feature.

khobbits commented:
Hmm, I still don't like this.
The idea of safe teleport is for 3 things:
- Stop you teleporting inside of blocks
- Stop you teleporting to your immediate death
- Stop you teleporting to people who are in the air

Case Study1:
User1 makes a /sethome then goes somewhere
User2 covers the home in lava
User1 types /home and dies

Case Study2:
User1 is flying or otherwise in the air, could be mid jump.
User2 teleports to User1 and dies

Case Study3:
User1 is swimming in lava with appropriate potion
User2 teleports to User1 and dies

There is a reason this code exists, the general consensuses is that people seem to be far more bothered about dying randomly, than people glitching through walls, generally.

Is there not a better alternative than simply removing the checks completely?

If the answer is still 'yes', then I'd honestly prefer Essentials is not connected to such reckless player killing.  I''m more than happy for you to use another plugin that doesn't have the same regard for users health.

Soz3r commented:
Would it be applicable to assign safe teleport to situational permission nodes?  This way, the administrators not only have the option to enable/disable but also assign who does.

> On Dec 11, 2013, at 6:18 AM, KHobbits notifications@github.com wrote:
> 
> Hmm, I still don't like this.
> The idea of safe teleport is for 3 things:
> 
> Stop you teleporting inside of blocks
> Stop you teleporting to your immediate death
> Stop you teleporting to people who are in the air
> Case Study1:
> User1 makes a /sethome then goes somewhere
> User2 covers the home in lava
> User1 types /home and dies
> 
> Case Study2:
> User1 is flying or otherwise in the air, could be mid jump.
> User2 teleports to User1 and dies
> 
> Case Study3:
> User1 is swimming in lava with appropriate potion
> User2 teleports to User1 and dies
> 
> There is a reason this code exists, the general consensuses is that people seem to be far more bothered about dying randomly, than people glitching through walls, generally.
> 
> Is there not a better alternative than simply removing the checks completely?
> 
> If the answer is still 'yes', then I'd honestly prefer Essentials is not connected to such reckless player killing. I''m more than happy for you to use another plugin that doesn't have the same regard for users health.
> 
> \
> Reply to this email directly or view it on GitHub.

drtshock commented:
I honestly think this is a good idea. I had to add it in the version that we use on my factions server. It's a way to "glitch" into bases that most servers don't allow. Since it is enabled by default with a big warning I don't see how adding the option could hurt anyone (lol).

fpigerre commented:
I definitely agree that such a feature should be added. In a lot of cases, not being able to teleport to a player who is flying over the void can be extremely frustrating. 

The comment in the config file should make sure that server owners are aware of the potential damage caused when the option to disable safe teleporting is enabled.

woodysgamertag commented:
This would be huge for factions players. Right now one of the primary methods for "glitch raiders" is to use /back, /tp, /tpa, etc. to a non-safe spot. Then the system looks for a safe place for the player and often that safe place is inside an enemy base. People lose days worth of work from a glitch raid and they are furious about it. It's actually a huge issue, it happens every single day on any busy server.

Because tnt doesn't work in water bases are often coated with several layers of water in factions so there are hundreds of opportunities to take advantage of this bug at any given time.

khobbits commented:
psgs: That is not an issue, if you have god and fly enabled, safe teleport is disabled automatically.  The same goes if you are in creative mode.

Commit:
Add the ability to disable teleport safety.
Requested by a few people - teleport safety isn't always a good thing for some servers.
Request is cancelled if not safe.

chrisgward commented:
Merged.

mrCaveman commented:
Wouldn't it be easier if you were to add an option to just make water listed as a material - like a door, or some tall grass? In that case, teleport glitches into vaults will not happen & safe teleport would work just fine (baring being teleported into water blocks)

Ceasel3ss commented:
This is really necessary, this is happening on multiple faction servers. This is allowing players to glitch into people's bases without any skill at all.
